torch/distributed/algorithms/ddp_comm_hooks/powerSGD_hook.py

```diff
@@ -242,6 +269,10 @@ def decompress(fut):
         for p, q, tensor in zip(ps, qs, high_rank_tensors):
             torch.matmul(p, q.t(), out=tensor)
             assert not torch.any(torch.isnan(tensor))
+
+        if state.use_error_feedback:
+            # Memorize the local errors.
+            state.error_dict[bucket_index] = input_tensor_cp - input_tensor
```

Comment:
Isn't the error input_tensor_cp - tensor, where `tensor` is the average approximation of `input_tensor`?

Reply:
`input_tensor` is already the average approximation. For each approximation tensor M = PQ, since both P and Q are already synchronized after 2 rounds of allreduce, M will also be exactly the same, and no need for another allreduce.